luavm/src/lib/mod.rs

```diff
@@ -90,19 +90,33 @@ mod tests {
         // vm.registers[0] has a reference to the _ENV variable
         // this is true because the compiler always loads the environment into register 0
         assert_eq!(
-            vm.registers[ENV_REG]
+            vm.registers.last().unwrap()[ENV_REG]
                 .get_attr(&LuaVal::from((String::from("x"), index_of_x)))
                 .unwrap(),
             LuaVal::new()
         );
         assert_eq!(vm.env_attrs[index_of_x], LuaVal::from(3));
         let index_of_y = 1;
         assert_eq!(
-            vm.registers[ENV_REG]
+            vm.registers.last().unwrap()[ENV_REG]
                 .get_attr(&LuaVal::from((String::from("y"), index_of_y)))
                 .unwrap(),
             LuaVal::new()
         );
         assert_eq!(vm.env_attrs[index_of_y], LuaVal::from(4));
     }
+
+    #[test]
+    fn function_call() {
+        // function f()
+        //     x = 3
+        //     local x = 4
+        // end
+        // f()
+        let mut vm = get_vm_for("function f()\n\tx = 3\n\tlocal x = 4\nend\nf()".to_string());
```

Comment:
You can use newlines in strings, so you can use the string to make sense of the program rather than the comment above.

Reply:
I never knew you could do this in Rust